AutoGraph/ResponseHandler.swift

```diff
@@ -53,7 +53,7 @@ open class ResponseHandler {
                     let object = try decoder.decode(SerializedObject.self, from: objectJson.encode())
                     
                     self.callbackQueue.addOperation {
-                        completion(.success(object, objectJson))
+                        completion(.success((object, objectJson)))
```

Comment:
@LeffelMania curious how you feel about this, think I should just make `AutoGraphResult` be an enum instead of a typealias?

Reply:
I might make it a `struct` instead, since it's already inside of a `.success` enum there aren't really multiple options for what the value will be, so my thought if we wanted to avoid this tuple stuff and be more flexible for the future would be something like
```swift
struct AutoGraphSuccessPayload<Value> {
    let value: Value
    let json: JSONValue
}

public typealias AutoGraphResult<Value> = Swift.Result<AutoGraphSuccessPayload<Value>, Error>
```
Different name probably. Happy to discuss, really depends on what problem we're trying to solve.

Reply:
I like where this is going. Maybe we can just have a tagging protocol that gets checked dynamically on the success value before being returned, if it's there we set the json as a property on it? Since json is sort of being iterated on how it's used as caching evolves anyway, this separates it from the base api.
```swift
public protocol JSONCarryable { var json: JSONValue { get set } }
```
?

Reply:
I think I'd need more clarification on what problem that's solving before having a real opinion. My main issue is that using a tuple across interfaces is kind of annoying, and using a struct solves that nuisance.

Reply:
what i'm saying is that we can make the json value a property on the returned structure arbitrarily, so our codegened types could have `.json` as a property globbed on to them.
```swift
/// Somewhere in codegen
struct MyOperation: JSONCarryable { var json: JSONValue ... }

/// Somewhere in client code
switch result {
    case .success(let value):
        /// value is the codegened GQL operation, but also has `.json` as a property if needed
        cache(value.json)
...
```
This keeps the api cleaner, i imagine we'll probably hide or remove that json exposure at some point as caching comes together and until then no other caller of autograph needs it anyway.

Reply:
good discussion though, got the right gears turning